code/game/objects/structures/morgue.dm

```diff
@@ -199,7 +199,23 @@ GLOBAL_LIST_EMPTY(crematoriums)
 		locked = 1
 		update_icon()
 
-		for(var/mob/living/M in contents)
+		if (!isnull(contents) && contents.len > 0)
+			cremateLoop(user, contents)
+
+		new /obj/effect/decal/cleanable/ash(src)
+		sleep(30)
+		if(!QDELETED(src))
+			locked = 0
+			update_icon()
+			playsound(src.loc, 'sound/machines/ding.ogg', 50, 1) //you horrible people
+
+// A recursive proc is needed for checking contents of contents, else bodies won't creamte in body bags
+/obj/structure/bodycontainer/crematorium/proc/cremateLoop(mob/user, list/conts)

```

Comment:
couldn't you loop through `GetAllContents() - src - tray` (assuming `tray` is the reference to the tray)?